Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -6302,9 +6302,12 @@ GC::Ptr<DOM::Position> Document::cursor_position() const
         return nullptr;
 
     Optional<HTML::FormAssociatedTextControlElement const&> target {};
-    if (is<HTML::HTMLInputElement>(*focused_element))
+    if (auto const* input_element = as_if<HTML::HTMLInputElement>(*focused_element)) {
+        // Some types of <input> tags shouldn't have a cursor, like buttons
+        if (!input_element->can_have_text_editing_cursor())
+            return nullptr;
         target = static_cast<HTML::HTMLInputElement const&>(*focused_element);
```

Comment:
This is what I mean. It works for me locally with your changes:
```suggestion
        target = *input_element;
```

Reply:
Oh alright, will do.